responders/src/UI/embc-responder/src/app/feature-components/wizard/step-ess-file/step-ess-file.service.ts

```diff
@@ -365,9 +352,12 @@ export class StepEssFileService {
   }
 
   public get needsIdentified(): string[] {
-    return Array.from(this.needs)
-    .map(need => this.loadEvacueeListService.getIdentifiedNeeds().find(value => value.value === need)?.description);
+    const getIdentifiedNeeds = this.loadEvacueeListService.getIdentifiedNeeds();
 
+    return Array.from(this.needs).map((need) => {
+      const foundNeed = getIdentifiedNeeds.find((value) => value.value === need);
+      return [IdentifiedNeed.ShelterAllowance, IdentifiedNeed.ShelterReferral].includes(foundNeed.value) ? 'Shelter' : foundNeed?.description;
```

Comment:
I think only shelter referral should be renamed shelter. Also, I'm sure there's a better way to centralize the caption resolution than the lame way I did it...

Reply:
Fixed it.